src/component/common/StringTruncator/StringTruncator.tsx

```diff
@@ -12,23 +14,53 @@ const StringTruncator = ({
     className,
     ...rest
 }: IStringTruncatorProps) => {
+    const calculateOverflowGenerator = useCallback(() => {
```

Comment:
Maybe we could use a simple `maxLength` prop instead of `maxWidth`? Seems like the end result would be similar.